### PR TITLE
Add note that documents YAML compatibility bug

### LINKED ISSUE #36526: Document YAML compatibility bug
**This is a Feature Request**

**What would you like to be added**
https://github.com/kubernetes/kubernetes/issues/34146 describes a bug in Kubernetes where YAML 1.2 documents are not parsed (Kubernetes does not have good support for YAML 1.1), but then also Kubernetes does incorrectly parse completely valid YAML 1.1 the way that the specification defines.

**Why is this needed**
The bug makes it hard for some people, using conformant libraries, to make YAML documents that Kubernetes accepts.
We should document the bug so that readers are aware of the shortcomings here.

**Comments**
/wg api-expression
/language en
/priority backlog

### PR DESCRIPTION
**Pull request by mrgiles**
Add warning note that documents Issue 34146, which addresses YAML 1.1/1.2 compatibility issues in Kubernetes.
Fixes #36526

### CODE REVIEW

**Comment by mrgiles** [comment]
@sftim this is a first draft, I'm not even sure if the content belongs to this document ( I asked you in the issue comments). Thanks

**Comment by mrgiles** [comment]
Does it make sense to you, @chrishiestand , @reinvantveer?

**Comment by reinvantveer** [comment]
> Does it make sense to you, @chrishiestand , @reinvantveer?

Perfect!

**Comment by chrishiestand** [comment]
> Does it make sense to you, @chrishiestand , @reinvantveer?

I'm not a fan of this documentation closing kubernetes/kubernetes#34146, however we're approaching 7 years since the issue was opened and if there's no other fix in sight then... okay? And since I haven't contributed anything between then and now my vote ought to count for very little.

I think it's a good start although perhaps not comprehensive enough because like @2rs2ts and @fraenkel said, there are incompatibilities between 1.1 and 1.2 other than booleans, such as scientific notation/string types. I am not aware of any exhaustive but concise list detailing breaking changes, but if one exists then we ought to link to it or copy it. I wasn't able to find one after some quick googling, however your google fu may be better.

So I think before this is merged you should urge avoidance of all incompatibilities between 1.1 and 1.2, even if we don't list them explicitly. In other words, we k8s users should try to stick to the syntax/grammar intersection of yamls 1.1. and 1.2.

Thanks for your help @mrgiles!

**Comment by fraenkel** [comment]
The list of changes can be found here:
https://yaml.org/spec/1.2.2/ext/changes/
See the `Changes in 1.2` section which describes what was removed/changed from 1.1

**Comment by sftim** [comment]
> I'm not a fan of this documentation closing [kubernetes/kubernetes#34146](https://github.com/kubernetes/kubernetes/issues/34146), however we're approaching 7 years since the issue was opened and if there's no other fix in sight then... okay? And since I haven't contributed anything between then and now my vote ought to count for very little.

@mrgiles I have edited the PR description to tweak that.

**Comment by mrgiles** [comment]
Thank you for the review, @chrishiestand. This PR was meant to close issue #36526, related to documentation, not the original [issue](https://github.com/kubernetes/kubernetes/issues/34146), related to the actual YAML issue.
 
As @fraenkel notes, the [YAML spec changes](https://spec.yaml.io/main/spec/1.2.2/ext/changes) doc lists the differences among versions. There are, indeed, other changes that I may add to the note or, I could just add the link as a reference for the reader to explore further? The change list is rather long. @sftim what do you suggest?

**Comment by sftim** [comment]
I wouldn't list all the differences between YAML versions here. We're documenting Kubernetes, not YAML itself.

**Comment by mrgiles** [comment]
@sftim I added a change to the note and squashed the commits. Thanks

**Comment by mrgiles** [comment]
@sftim do you have any feedback on the latest change? How should I proceed? Thanks

**Comment by mrgiles** [comment]
@reylejano can you take a look at this PR? Thanks

File: content/en/docs/concepts/overview/working-with-objects/_index.md
[the remark this answers is not in the document]
Comment (mrgiles) [reply content/en/docs/concepts/overview/working-with-objects/_index.md:1]: Thanks for the advice @sftim. 
Do I need to open a separate PR to modify the configuration/overview page or can I do both pages here?

[the remark this answers is not in the document]
Comment (mrgiles) [reply content/en/docs/concepts/overview/working-with-objects/_index.md:1]: I moved the note over to the other document and left a link in its place, as you suggested @sftim. 
Does it look okay to you @nate-double-u? 
Thanks!

File: content/en/docs/concepts/configuration/overview.md [hunk 1]
```diff
@@ -37,6 +37,23 @@ to others, please don't hesitate to file an issue or submit a PR.
 
 - Put object descriptions in annotations, to allow better introspection.
 
+{{< note >}}
+There is is an incompatibility between [YAML 1.2](https://yaml.org/spec/1.2.0/#id2602744) and
```
[the remark this answers is not in the document]
Comment (mrgiles) [reply content/en/docs/concepts/configuration/overview.md]: Done. Thanks for the feedback @stormqueen1990

File: content/en/docs/concepts/configuration/overview.md [hunk 2]
```diff
@@ -37,6 +37,23 @@ to others, please don't hesitate to file an issue or submit a PR.
 
 - Put object descriptions in annotations, to allow better introspection.
 
+{{< note >}}
+There is is a breaking change introduced in the [YAML 1.2](https://yaml.org/spec/1.2.0/#id2602744)
```
Comment (stormqueen1990) [inline content/en/docs/concepts/configuration/overview.md:41]:
```suggestion
There is a breaking change introduced in the [YAML 1.2](https://yaml.org/spec/1.2.0/#id2602744)
```
Small typo!

**Review by stormqueen1990 — COMMENTED** [review]
Mostly LGTM, just a minor typo found!

File: content/en/docs/concepts/configuration/overview.md (hunk 2 above)
Comment (mrgiles) [reply content/en/docs/concepts/configuration/overview.md]: Thank you @stormqueen1990 
Can't believe I rephrased the paragraph and repeated the original typo :)

**Review by stormqueen1990 — COMMENTED** [review]
/lgtm

File: content/en/docs/concepts/overview/working-with-objects/_index.md [hunk 3]
```diff
@@ -120,6 +120,11 @@ satisfy the StatefulSet specification.
 Different kinds of object can also have different `.status`; again, the API reference pages
 detail the structure of that `.status` field, and its content for each different type of object.
 
+{{< note >}}
+See [Configuration Best Practices](/docs/concepts/configuration/overview/) for additional
+information on writing YAML configuration files.
```
Comment (sftim) [inline content/en/docs/concepts/overview/working-with-objects/_index.md:125]:
```suggestion
information on writing manifests in YAML.
```
(maybe with a tooltip for manifest)

**Review by sftim — COMMENTED** [review]
/approve

**Pull request merged**